### PR TITLE
Fix build warnings

### PR DESCRIPTION
**Pull request by jlowe**
This fixes a number of warnings in the build that have accumulated recently.  Specifically these warnings are addressed:
```
[WARNING] Some problems were encountered while building the effective model for com.nvidia:rapids-4-spark-integration-tests_2.12:jar:0.3.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.(groupId:artifactId)' must be unique but found duplicate declaration of plugin org.apache.maven.plugins:maven-assembly-plugin @ line 129, column 21
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[...]
[WARNING] /home/jlowe/src/rapids-plugin-4-spark/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala:306: match may not be exhaustive.
It would fail on the following inputs: (??, _), (ColumnVector(), _), (Scalar(), Scalar()), (Scalar(), _), (_, ??), (_, ColumnVector()), (_, Scalar()), (_, _)
[WARNING]   private[this] def combineButNoClose(r: Any, c: Any): Any = (r, c) match {
[WARNING]                                                              ^
[WARNING] /home/jlowe/src/rapids-plugin-4-spark/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala:361: match may not be exhaustive.
It would fail on the following inputs: (??, _), (ColumnVector(), _), (Scalar(), Scalar()), (Scalar(), _), (_, ??), (_, ColumnVector()), (_, Scalar()), (_, _)
[WARNING]   private[this] def combineButNoCloseFp(r: Any, c: Any): Any = (r, c) match {
[...]
[WARNING] testSourceDirectory is not specified or does not exist value=/home/jlowe/src/rapids-plugin-4-spark/sql-plugin/src/test/scala
[...]
[WARNING] testSourceDirectory is not specified or does not exist value=/home/jlowe/src/rapids-plugin-4-spark/shuffle-plugin/src/test/scala
[...]
[WARNING] sourceDirectory is not specified or does not exist value=/home/jlowe/src/rapids-plugin-4-spark/shims/src/main/scala
[WARNING] testSourceDirectory is not specified or does not exist value=/home/jlowe/src/rapids-plugin-4-spark/shims/src/test/scala
[...]
[WARNING] testSourceDirectory is not specified or does not exist value=/home/jlowe/src/rapids-plugin-4-spark/shims/spark300/src/test/scala
[...]
[WARNING] testSourceDirectory is not specified or does not exist value=/home/jlowe/src/rapids-plugin-4-spark/shims/spark300emr/src/test/scala
[...]
[WARNING] testSourceDirectory is not specified or does not exist value=/home/jlowe/src/rapids-plugin-4-spark/shims/spark301/src/test/scala
[...]
[WARNING] testSourceDirectory is not specified or does not exist value=/home/jlowe/src/rapids-plugin-4-spark/shims/spark301emr/src/test/scala
[...]
[WARNING] testSourceDirectory is not specified or does not exist value=/home/jlowe/src/rapids-plugin-4-spark/shims/spark310/src/test/scala
[...]
[WARNING] testSourceDirectory is not specified or does not exist value=/home/jlowe/src/rapids-plugin-4-spark/shims/spark302/src/test/scala
[...]
[WARNING] sourceDirectory is not specified or does not exist value=/home/jlowe/src/rapids-plugin-4-spark/shims/aggregator/src/main/scala
[WARNING] testSourceDirectory is not specified or does not exist value=/home/jlowe/src/rapids-plugin-4-spark/shims/aggregator/src/test/scala
[...]
[WARNING] sourceDirectory is not specified or does not exist value=/home/jlowe/src/rapids-plugin-4-spark/tests/src/main/scala
[...]
/home/jlowe/src/rapids-plugin-4-spark/integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/CompareResults.scala:31: warning: Variable SPARK undefined in comment for object CompareResults in object CompareResults
/home/jlowe/src/rapids-plugin-4-spark/integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/CompareResults.scala:31: warning: Variable SPARK undefined in comment for object CompareResults in object CompareResults
/home/jlowe/src/rapids-plugin-4-spark/integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/CompareResults.scala:31: warning: Variable CUDF undefined in comment for object CompareResults in object CompareResults
/home/jlowe/src/rapids-plugin-4-spark/integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/CompareResults.scala:34: warning: Variable SPARK undefined in comment for object CompareResults in object CompareResults
```

### CODE REVIEW

**Comment by jlowe** [comment]
build

File: pom.xml [hunk 1]
```diff
@@ -439,7 +439,7 @@
                     <configuration>
                         <verbose>false</verbose>
                         <failOnViolation>true</failOnViolation>
-                        <includeTestSourceDirectory>true</includeTestSourceDirectory>
+                        <includeTestSourceDirectory>false</includeTestSourceDirectory>
```
Comment (revans2) [inline pom.xml:442]: Why do we have to turn it off here and in the child pom.xml files?

Comment (jlowe) [reply pom.xml:442]: Updated the child pom.xml files to remove the redundant settings.

**Comment by jlowe** [comment]
build

**Review by revans2 — APPROVED** [review]

**Pull request merged**